ui/raidboss/popup-text.js

```diff
@@ -770,8 +775,7 @@ class PopupText {
           text = triggerUpperCase(text);
         let holder = this[lowerTextKey].getElementsByClassName('holder')[0];
         let div = this._makeTextElement(text, textElementClass);
-        this._addText(holder, div);
-        this._scheduleRemoveText(holder, div,
+        this._addText(holder, div,
```

Comment:
Also, do you think it would make more sense to pass lowerTextKey instead of `holder`, since `holder` won't even exist for the emulator? I don't feel strongly here, but just curious if you think that makes more sense for this function.

Reply:
Hmm, maybe this makes sense?

```javascript
  _createTextFor(text, textType, lowerTextKey, duration) {
    // info-text
    let textElementClass = textType + '-text';
    if (textType !== 'info')
      text = triggerUpperCase(text);
    let holder = this[lowerTextKey].getElementsByClassName('holder')[0];
    let div = this._makeTextElement(text, textElementClass);
    this._addText(holder, div, duration);
  }

  _addText(container, e, duration) {
    container.appendChild(e);
    if (container.children.length > this.kMaxRowsOfText)
      container.removeChild(container.children[0]);

    this._scheduleRemoveText(container, div, duration);
  }

  _addTextFor(textType, triggerHelper) {
    // Info
    let textTypeUpper = textType[0].toUpperCase() + textType.slice(1);
    // infoText
    let lowerTextKey = textType + 'Text';
    // InfoText
    let upperTextKey = textTypeUpper + 'Text';
    let textObj = triggerHelper.triggerOptions[upperTextKey] ||
      triggerHelper.trigger[lowerTextKey] || triggerHelper.response[lowerTextKey];
    if (textObj) {
      let text = triggerHelper.valueOrFunction(textObj);
      triggerHelper.defaultTTSText = triggerHelper.defaultTTSText || text;
      if (text && triggerHelper.textAlertsEnabled) {
        this._createTextFor(text, textType, lowerTextKey,
          (triggerHelper.duration.fromTrigger || triggerHelper.duration[lowerTextKey]));
        if (!triggerHelper.soundUrl) {
          triggerHelper.soundUrl = this.options[textTypeUpper + 'Sound'];
          triggerHelper.soundVol = this.options[textTypeUpper + 'SoundVolume'];
        }
      }
    }
  }
```

Reply:
Sounds good!  Would the raid emulator use `_addText` or could `_addText` be rolled up into `_createTextFor` and raid emulator can override that one thing?

Reply:
Can roll it up, but at that point might as well roll up _scheduleRemoveText as well:

```javascript
  _createTextFor(text, textType, lowerTextKey, duration) {
    // info-text
    let textElementClass = textType + '-text';
    if (textType !== 'info')
      text = triggerUpperCase(text);
    let holder = this[lowerTextKey].getElementsByClassName('holder')[0];
    let div = this._makeTextElement(text, textElementClass);
    this._addText(holder, div, duration);
    holder.appendChild(e);
    if (holder.children.length > this.kMaxRowsOfText)
      holder.removeChild(holder.children[0]);

    window.setTimeout(() => {
      for (let i = 0; i < holder.children.length; ++i) {
        if (holder.children[i] == e) {
          holder.removeChild(e);
          break;
        }
      }
    }, duration * 1000);
  }

  _addTextFor(textType, triggerHelper) {
    // Info
    let textTypeUpper = textType[0].toUpperCase() + textType.slice(1);
    // infoText
    let lowerTextKey = textType + 'Text';
    // InfoText
    let upperTextKey = textTypeUpper + 'Text';
    let textObj = triggerHelper.triggerOptions[upperTextKey] ||
      triggerHelper.trigger[lowerTextKey] || triggerHelper.response[lowerTextKey];
    if (textObj) {
      let text = triggerHelper.valueOrFunction(textObj);
      triggerHelper.defaultTTSText = triggerHelper.defaultTTSText || text;
      if (text && triggerHelper.textAlertsEnabled) {
        this._createTextFor(text, textType, lowerTextKey,
          (triggerHelper.duration.fromTrigger || triggerHelper.duration[lowerTextKey]));
        if (!triggerHelper.soundUrl) {
          triggerHelper.soundUrl = this.options[textTypeUpper + 'Sound'];
          triggerHelper.soundVol = this.options[textTypeUpper + 'SoundVolume'];
        }
      }
    }
  }
```

Reply:
Looks good to me!